bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -2511,7 +2511,12 @@ private static boolean checkIsLikeArrayType(Object sourceValue, BArrayType targe
             }
         }
 
-        for (int i = 0; i < source.size(); i++) {
+        int targetTypeSize = targetType.getSize();
+        int sourceSize = source.size();
+        if ((targetTypeSize != -1) && (sourceSize != targetTypeSize)) {
```

Comment:
Do we need -1 check here?

Reply:
Need to consider the array size, only when the targetType array is a closed one (i.e. `[2, 2, 3] is int[]` is `true`). When it is an open one, its size is given as -1.

Reply:
Shall we use `targetType.getState() != ArrayState.OPEN` instead of the -1 check?

Reply:
Addressed with e29f827.